daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs

```diff
@@ -674,35 +678,24 @@ runStart sandboxPortM (StartNavigator shouldStartNavigator) (OpenBrowser shouldO
             when shouldWaitForSignal $
                 void $ race (waitExitCode navigatorPh) (waitExitCode sandboxPh)
 
-    where navigatorPort = NavigatorPort 7500
-          withNavigator' sandboxPh =
-              if shouldStartNavigator
-                  then withNavigator
-                  else (\_ _ _ f -> f sandboxPh)
+    where
+        navigatorPort = NavigatorPort 7500
+        defaultSandboxPort = SandboxPort 6865
+        withNavigator' sandboxPh =
+            if shouldStartNavigator
+                then withNavigator
+                else (\_ _ _ f -> f sandboxPh)
 
 data HostAndPortFlags = HostAndPortFlags { hostM :: Maybe String, portM :: Maybe Int }
 
 getHostAndPortDefaults :: HostAndPortFlags -> IO HostAndPort
 getHostAndPortDefaults HostAndPortFlags{hostM,portM} = do
     host <- fromMaybeM getProjectLedgerHost hostM
     port <- fromMaybeM getProjectLedgerPort portM
-    return HostAndPort {host,port}
-
-runListParties :: HostAndPortFlags -> IO ()
-runListParties flags = do
-    hp <- getHostAndPortDefaults flags
-    putStrLn $ "Listing parties at " <> show hp
-    xs <- Ledger.listParties hp
-    if null xs then putStrLn "no parties are known" else mapM_ print xs
-    exitSuccess
+    return HostAndPort {..}
 
-runAllocateParty :: HostAndPortFlags -> String -> IO ()
-runAllocateParty flags name = do
-    hp <- getHostAndPortDefaults flags
-    putStrLn $ "Checking party allocation at " <> show hp
-    allocatePartyIfRequired hp name
-    exitSuccess
 
+-- | Allocate project parties and upload project DAR file to ledger.
 runDeploy :: HostAndPortFlags -> IO ()
 runDeploy flags = do
     hp <- getHostAndPortDefaults flags
```

Comment:
Should `runDeploy` call `runLedgerAllocateParty` and `runLedgerUploadDar`? It seems like this has a risk of the two getting slightly out of sync which could get rather confusing.

Reply:
👍 I'll save that refactoring for another PR, because the behavior is already a little bit different wrt allocate parties, and this seems like a good idea.